### ISSUE TITLE
Test failures observed with AccessDeniedException error

### ISSUE THREAD

**Issue by bhavenamrata** [elastic/curator#1193]
### Expected Behavior
Tests should pass

### Actual Behavior
11 tests failing with error `failed to create blob container`

### Steps to Reproduce the Problem
Clone v5.5.1 of Curator and run `python setup.py test`

### Specifications

  - Version: v5.5.1
  - Platform: Intel x86_64
  - ES version: v5.6.1

### Detailed Description
While running the tests, 11 failing due to `AccessDeniedException`. Please check below log for further details.
I have built Elasticsearch from source (v5.6.1 as well as v5.2.1) , both show same error.
Also have provided full rights to `/tmp`. 
```
======================================================================
ERROR: test_restore_wildcard (test.integration.test_restore.TestCLIRestore)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/test1/curator/test/integration/test_restore.py", line 144, in test_restore_wildcard
    self.create_snapshot(snap_name, ','.join(indices))
  File "/home/test1/curator/test/integration/__init__.py", line 142, in create_snapshot
    self.create_repository()
  File "/home/test1/curator/test/integration/__init__.py", line 150, in create_repository
    self.client.snapshot.create_repository(repository=self.args['repository'], body=body)
  File "build/bdist.linux-x86_64/egg/elasticsearch/client/utils.py", line 73, in _wrapped
    return func(*args, params=params, **kwargs)
  File "build/bdist.linux-x86_64/egg/elasticsearch/client/snapshot.py", line 108, in create_repository
    repository), params=params, body=body)
  File "build/bdist.linux-x86_64/egg/elasticsearch/transport.py", line 312, in perform_request
    status, headers, data = connection.perform_request(method, url, params, body, ignore=ignore, timeout=timeout)
  File "build/bdist.linux-x86_64/egg/elasticsearch/connection/http_urllib3.py", line 129, in perform_request
    self._raise_error(response.status, raw_data)
  File "build/bdist.linux-x86_64/egg/elasticsearch/connection/base.py", line 125, in _raise_error
    raise HTTP_EXCEPTIONS.get(status_code, TransportError)(status_code, error_message, additional_info)
TransportError: TransportError(500, u'exception', u'failed to create blob container')

```
```
test_restore_wildcard (test.integration.test_restore.TestCLIRestore) ... [2018-03-30T10:58:16,427][INFO ][o.e.c.m.MetaDataCreateIndexService] [2mtcG9i] [my_index1] creating index, cause [auto(index api)], templates [], shards [5]/[1], mappings []
.
.
[2018-03-30T10:58:22,726][WARN ][r.suppressed             ] path: /_snapshot/TEST_REPOSITORY, params: {repository=TEST_REPOSITORY}
org.elasticsearch.ElasticsearchException: failed to create blob container
        at org.elasticsearch.common.blobstore.fs.FsBlobStore.blobContainer(FsBlobStore.java:70) ~[elasticsearch-5.2.1-SNAPSHOT.jar:5.2.1-SNAPSHOT]
        at org.elasticsearch.repositories.blobstore.BlobStoreRepository.startVerification(BlobStoreRepository.java:723) ~[elasticsearch-5.2.1-SNAPSHOT.jar:5.2.1-SNAPSHOT]
        at org.elasticsearch.repositories.RepositoriesService.verifyRepository(RepositoriesService.java:210) [elasticsearch-5.2.1-SNAPSHOT.jar:5.2.1-SNAPSHOT]
        at org.elasticsearch.repositories.RepositoriesService$VerifyingRegisterRepositoryListener.onResponse(RepositoriesService.java:413) [elasticsearch-5.2.1-SNAPSHOT.jar:5.2.1-SNAPSHOT]
        at org.elasticsearch.repositories.RepositoriesService$VerifyingRegisterRepositoryListener.onResponse(RepositoriesService.java:398) [elasticsearch-5.2.1-SNAPSHOT.jar:5.2.1-SNAPSHOT]
        at org.elasticsearch.cluster.AckedClusterStateUpdateTask.onAllNodesAcked(AckedClusterStateUpdateTask.java:64) [elasticsearch-5.2.1-SNAPSHOT.jar:5.2.1-SNAPSHOT]
        at org.elasticsearch.cluster.service.ClusterService$SafeAckedClusterStateTaskListener.onAllNodesAcked(ClusterService.java:1047) [elasticsearch-5.2.1-SNAPSHOT.jar:5.2.1-SNAPSHOT]
        at org.elasticsearch.cluster.service.ClusterService$AckCountDownListener.onNodeAck(ClusterService.java:1305) [elasticsearch-5.2.1-SNAPSHOT.jar:5.2.1-SNAPSHOT]
        at org.elasticsearch.cluster.service.ClusterService$DelegetingAckListener.onNodeAck(ClusterService.java:1245) [elasticsearch-5.2.1-SNAPSHOT.jar:5.2.1-SNAPSHOT]
        at org.elasticsearch.cluster.service.ClusterService.publishAndApplyChanges(ClusterService.java:828) [elasticsearch-5.2.1-SNAPSHOT.jar:5.2.1-SNAPSHOT]
        at org.elasticsearch.cluster.service.ClusterService.runTasks(ClusterService.java:628) [elasticsearch-5.2.1-SNAPSHOT.jar:5.2.1-SNAPSHOT]
        at org.elasticsearch.cluster.service.ClusterService$UpdateTask.run(ClusterService.java:1112) [elasticsearch-5.2.1-SNAPSHOT.jar:5.2.1-SNAPSHOT]
        at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:527) [elasticsearch-5.2.1-SNAPSHOT.jar:5.2.1-SNAPSHOT]
        at org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.runAndClean(PrioritizedEsThreadPoolExecutor.java:238) [elasticsearch-5.2.1-SNAPSHOT.jar:5.2.1-SNAPSHOT]
        at org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.run(PrioritizedEsThreadPoolExecutor.java:201) [elasticsearch-5.2.1-SNAPSHOT.jar:5.2.1-SNAPSHOT]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_151]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_151]
        at java.lang.Thread.run(Thread.java:748) [?:1.8.0_151]
Caused by: java.nio.file.AccessDeniedException: /tmp/tmpwewU0QXMG8Z70E/tests-y6Rs4rELR8W5RDUI00ssPA
        at sun.nio.fs.UnixException.translateToIOException(UnixException.java:84) ~[?:?]
        at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:102) ~[?:?]
        at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:107) ~[?:?]
        at sun.nio.fs.UnixFileSystemProvider.createDirectory(UnixFileSystemProvider.java:384) ~[?:?]
        at java.nio.file.Files.createDirectory(Files.java:674) ~[?:1.8.0_151]
        at java.nio.file.Files.createAndCheckIsDirectory(Files.java:781) ~[?:1.8.0_151]
        at java.nio.file.Files.createDirectories(Files.java:767) ~[?:1.8.0_151]
        at org.elasticsearch.common.blobstore.fs.FsBlobStore.buildAndCreate(FsBlobStore.java:86) ~[elasticsearch-5.2.1-SNAPSHOT.jar:5.2.1-SNAPSHOT]
        at org.elasticsearch.common.blobstore.fs.FsBlobStore.blobContainer(FsBlobStore.java:68) ~[elasticsearch-5.2.1-SNAPSHOT.jar:5.2.1-SNAPSHOT]
        ... 17 more
ERROR
.
.
```

Any pointers to resolve this would be helpful.

**Comment by untergeek** [comment]
I wish I could help you here, but somehow, on your system, _Elasticsearch itself_ is unable to read/write from the specified directory, even in `/tmp` (hence the `500` error from Elasticsearch, evident in `TransportError: TransportError(500, u'exception', u'failed to create blob container')`). 

Solve _Elasticsearch's_ access problems, and the Curator test will work just fine.  Is Elasticsearch running as a different user than the user executing Curator?  That alone could explain the issue.

**Comment by untergeek** [comment]
You've specified the `path.repo` option for your test cluster, right?

In my test Elasticsearch instances (both local, and Travis CI), I specify: 

```
path.repo: /
```
in the elasticsearch.yml file.

**Comment by bhavenamrata** [comment]
@untergeek, thanks for your reply.

Initially the tests failed with below error: 
``` 
TransportError: TransportError(500, u'repository_exception', u"[TEST_REPOSITORY] location [/tmp/tmp0tlhNbAFKPCKY3] doesn't match any of the locations specified by path.repo because this setting is empty") 
```

Hence I updated `path.repo: /tmp` in `elasticsearch.yml`, which started giving `AccessDeniedException`.

Elasticsearch and Curator tests are being run by the same user. Also I tried giving all permissions to `/tmp` along with changing the owner of `/tmp` to the user. However, its not helping.
Am I missing some other configuration in Elasticsearch?
